synapse/models/infotech.py

```diff
@@ -25,559 +24,770 @@ class SemverType(DataType):
-                    'subof': 'str',
-                    'doc': 'A developer-selected string.'}),
-
-                ('it:dev:int', {
-                    'subof': 'int',
-                    'doc': 'A developer-selected integer constant.'}),
-
-                ('it:exec:proc', {
-                    'subof': 'guid',
-                    'doc': 'A process executing on a host. May be an actual (e.g., endpoint) or virtual (e.g., malware sandbox) host.'}),
-
-                ('it:dev:pipe', {
-                    'subof': 'it:dev:str',
-                    'doc': 'A string representing a named pipe.'}),
-
-                ('it:dev:mutex', {
-                    'subof': 'it:dev:str',
-                    'doc': 'A string representing a mutex.'}),
-
-                ('it:dev:regkey', {
-                    'subof': 'it:dev:str',
-                    'lower': 1,
+                    'ex': 'cve-2012-0158'
+                }),
+                ('it:dev:str', ('str', {}), {
+                    'doc': 'A developer-selected string.'
+                }),
+                ('it:dev:pipe', ('str', {}), {
+                    'doc': 'A string representing a named pipe.',
+                }),
+                ('it:dev:mutex', ('str', {}), {
+                    'doc': 'A string representing a mutex.',
+                }),
+                ('it:dev:int', ('int', {}), {
+                    'doc': 'A developer selected integer constant.',
+                }),
+                ('it:dev:regkey', ('str', {}), {
                     'doc': 'A Windows registry key.',
-                    'ex': 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Run'}),
-
-                ('it:dev:regval', {
-                    'subof': 'comp',
-                    'fields': 'key=it:dev:regkey',
-                    'optfields': 'str=it:dev:str,int=it:dev:int,bytes=file:bytes',
-                    'doc': 'A Windows registry key/value pair.'}),
-
-                ('it:semver', {
-                    'ctor': 'synapse.models.infotech.SemverType',
-                    'doc': 'Semantic Version type.'}),
-
-                ('it:prod:soft', {
-                    'subof': 'guid',
-                    'doc': 'A arbitrary, unversioned software product.'}),
-
-                ('it:prod:softver', {
-                    'subof': 'guid',
-                    'doc': 'A version of a particular software product.'}),
-
-                ('it:hostsoft', {
-                    'subof': 'comp',
-                    'fields': 'host,it:host|softver,it:prod:softver',
-                    'doc': 'A version of a software product which is present on a given host.'}),
-
-                ('it:auth:passwdhash', {
-                    'subof': 'guid',
-                    'ex': '(hash:md5=17d3533fba2669f84a225a9a04caa783)'}),
+                    'ex': 'HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Run',
+                }),
+                ('it:dev:regval', ('guid', {}), {
```

Comment:
This used to be a comp of `it:dev:regkey`, should it be a sub of`it:dev:regkey`?

Reply:
Nevermind, didn't see the optfields